app/src/main/java/org/greenstand/android/TreeTracker/fragments/LoginFragment.kt

```diff
@@ -0,0 +1,200 @@
+//Here we will watch the phone edit text to check if the input number(#phoneNumberInput) is in our database already.
+    //if the #phoneNumberInput is in our database we will turn Login button to ActiveButtonStyle and we will set
+    // ClickListener on this button to go on further with the login feature
+     v.phoneEditTextLogin.addTextChangedListener(object: TextWatcher {
+        @SuppressLint("NewApi")
+        override fun afterTextChanged(p0: Editable?) {
+            phoneNumberEntered = p0.toString()
+            if(isExistingUser(phoneNumberEntered!!)) {
+                activateLoginButton()
+            }else {
+                v.emailEditText.addTextChangedListener(object: TextWatcher {
+                    @SuppressLint("NewApi")
+                    override fun afterTextChanged(p0: Editable?) {
+                        emailEntered = p0.toString()
+                        if (isExistingUser(emailEntered!!) || isExistingUser(phoneNumberEntered!!)) {
+                            activateLoginButton()
+                        }else
+                            when (isExistingUser(emailEntered!!)) {
+                                true -> activateLoginButton()
+                                else -> {
+                                    sign_up_button?.visibility = View.VISIBLE
+                                }
+                            }
+                    }
+
+                    override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int){
+                        inactivateLoginButton()
+                    }
+
+                    override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                        inactivateLoginButton()
+                    }
+                })
+                when (isExistingUser(phoneNumberEntered!!)) {
+                    true -> activateLoginButton()
+                    else -> {
+                        sign_up_button?.visibility = View.VISIBLE
+                    }
+                }
+
+            }
+
+        }
+
+        override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            inactivateLoginButton()
+        }
+
+        override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            inactivateLoginButton()
+        }
+
+    })
+
+    v.sign_up_button.setOnClickListener(object : View.OnClickListener{
+        override fun onClick(v: View?) {
+
+
+        }
+
+    })
+
+ return v
+}
+
+
+//This is the method that will check into DB if the textEntered(that could be the phoneNumber or the email) already
+// exists in our DB. If exists it will return TRUE else it will return FALSE. in fact, it will return the if the
+// planterId is different than -1 ( this is the initial value given to var planterId), this var will receive
+// the planterId from the DB if the user already exists
+private fun isExistingUser(textEntered: String): Boolean {
+    runBlocking {
```

Comment:
This function is blocking the main thread, this means any UI has to wait for this to finish. In this case it's not too bad, but in general we should not be doing this.
One way to resolve this to to have the area this is called, call it inside of a `launch` function. This function can than be made a suspending function, and the runBlocking and async can be removed.

Reply:
It seems that I misunderstood at first the usage of runBlocking. I will read more about this and about launch too. Thanks!